src/main/java/fr/esgi/al5_2/Tayarim/services/AdministrateurService.java

```diff
@@ -0,0 +1,171 @@
+    return AdministrateurMapper.entityToDto(optionalAdministrateur.get());
+  }
+
+  public AdministrateurDTO getAdministrateurByEmail(@NonNull String email) {
+    Optional<Administrateur> optionalAdministrateur = administrateurRepository.findFirstByEmail(
+        email);
+    if (optionalAdministrateur.isEmpty()) {
+      throw new AdministrateurNotFoundException();
+    }
+
+    return AdministrateurMapper.entityToDto(optionalAdministrateur.get());
+  }
+
+  @Transactional
+  public AdministrateurDTO updateAdministrateur(@NonNull Long id,
+      @NonNull AdministrateurUpdateDTO administrateurUpdateDTO) {
+
+    if (
+        administrateurUpdateDTO.getNom() == null
+            && administrateurUpdateDTO.getPrenom() == null
+            && administrateurUpdateDTO.getEmail() == null
+            && administrateurUpdateDTO.getNumTel() == null
+            && administrateurUpdateDTO.getMotDePasse() == null
+    ) {
+      throw new AdministrateurInvalidUpdateBody();
+    }
+
+    Optional<Administrateur> optionalAdministrateur = administrateurRepository.findById(id);
+    if (optionalAdministrateur.isEmpty()) {
+      throw new AdministrateurNotFoundException();
+    }
+
+    if (administrateurUpdateDTO.getEmail() != null && administrateurRepository.findFirstByEmail(
+        administrateurUpdateDTO.getEmail()).isPresent()) {
+      throw new ProprietaireInvalidUpdateBody();
+    }
+
+    if (administrateurUpdateDTO.getNumTel() != null && administrateurRepository.findFirstByNumTel(
+        administrateurUpdateDTO.getNumTel()).isPresent()) {
+      throw new ProprietaireInvalidUpdateBody();
+    }
+
+    Administrateur administrateur = optionalAdministrateur.get();
+
+    administrateur.setNom(
+        administrateurUpdateDTO.getNom() != null ? administrateurUpdateDTO.getNom()
+            : administrateur.getNom());
+    administrateur.setPrenom(
+        administrateurUpdateDTO.getPrenom() != null ? administrateurUpdateDTO.getPrenom()
+            : administrateur.getPrenom());
+    administrateur.setEmail(
+        administrateurUpdateDTO.getEmail() != null ? administrateurUpdateDTO.getEmail()
+            : administrateur.getEmail());
+    administrateur.setNumTel(
+        administrateurUpdateDTO.getNumTel() != null ? administrateurUpdateDTO.getNumTel()
+            : administrateur.getNumTel());
+    administrateur.setMotDePasse(administrateurUpdateDTO.getMotDePasse() != null ? hashPassword(
+        administrateurUpdateDTO.getMotDePasse()) : administrateur.getMotDePasse());
+
+    return AdministrateurMapper.entityToDto(administrateurRepository.save(administrateur));
+  }
+
+  @Transactional
+  public AdministrateurDTO deleteAdministrateur(@NonNull Long id) {
+
+    Optional<Administrateur> optionalAdministrateur = administrateurRepository.findById(id);
+    if (optionalAdministrateur.isEmpty()) {
+      throw new AdministrateurNotFoundException();
+    }
+
+    AdministrateurDTO administrateurDTO = AdministrateurMapper.entityToDto(
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'administrateurDTO' must contain no more than '1' consecutive capital letters.